### PR TITLE
Plugin respects Jenkins proxy settings

### PR DESCRIPTION
**Pull request by ml-extern**
With these additions, the plugin will respect the proxy settings in Jenkins and configure the http connection accordingly.

This allows the user to address servers both within and outside the local area network and still use this plugin.

### CODE REVIEW

**Comment by tbores** [comment]
We would also need this change. When could be the change merged?

File: src/main/java/hockeyapp/HockeyappRecorder.java [hunk 1]
```diff
@@ -169,10 +190,21 @@ private HttpClient createPreconfiguredHttpClient() {
             httpclient.getParams().setParameter(ConnRoutePNames.DEFAULT_PROXY, proxy);
         }
 
+        /// Logging output ////////////////
```
Comment (Brantone) [inline src/main/java/hockeyapp/HockeyappRecorder.java]: The white space formatting here is inconsistent with rest of the file / project. Please clean it up.

File: src/main/java/hockeyapp/HockeyappRecorder.java [hunk 2]
```diff
@@ -146,13 +147,33 @@ public boolean isDebugEnabled() {
...
+                Pattern noproxypattern = noProxyHostPatterns.get(i);
+                if (noproxypattern.matcher(url.getHost()).matches()) {
+                    useProxy = false;
+                    matchedPattern = noproxypattern.toString();
+                };
+            }
+        }
+
+        // Proxy setting, we have a Proxy _and_ the provided URL does not match any no-proxy-override
+        if (hasProxy &&
```
Comment (Brantone) [inline src/main/java/hockeyapp/HockeyappRecorder.java]: Why split onto 2 lines?

**Comment by dannascimento** [comment]
We would also need this change. When could be the change merged?

**Comment by jjmiv** [comment]
is this related to the java socket timeouts and requiring a proxy login?

what's the status of this getting merged? let me know if i can assist.

**Comment by mezpahlan** [comment]
@jjmiv Apologies for the late reply. I became a maintainer well after this PR was raised. As best as I can tell it's to respect a proxy if it is set and to better handle edge cases related to various proxy settings.

The automatic checks tell me there are conflicts that need resolving. I'll have a look at those on the weekend and see about getting this merged. 

Thanks for the offer of help. I'll reach out if I need any assistance.

p.s. I *think* there was a very similar PR merged in the last release that also aimed to resolve proxy issues. If so that may supercede this one. I'll find out. Could be we need both?

**Comment by showpony** [comment]
I've been having trouble using this jenkins plugin to upload builds to hockey - I'm getting `java.net.SocketTimeoutException: Read timed out` errors.  I think it has to do with my company's proxy setup-  Hopefully this PR will fix that! *crossing fingers*

**Comment by mezpahlan** [comment]
Hi @jjmiv, @showpony apologies both had a tough couple weeks at work not really had time to look back into this. Thanks for the offer of help. Here's the status as I see it.

At present, the PR is failing some checks. Probably because the master branch has moved on quite a bit since this was raised. I'll see what I can do to update this PR and bring it into line.

Company proxy set ups! Tell me about it! This *may* help but just in case it doesn't can you list here what exactly the issue is with your company proxy set up. Could need a separate bug raised for the plugin or (best case) this resolves the issue.

**Comment by mezpahlan** [comment]
Hmm...... seems in a wacky state. I tried a simple conflict resolution that got further than before but the check on the server are still failing.

**Pull request merged**